app/controllers/concerns/secured.rb

```diff
@@ -2,23 +2,11 @@ module Secured
   extend ActiveSupport::Concern
 
   included do
-    before_action :logged_in_using_omniauth?
+    before_action :redirect_unauthenticated
```

Comment:
Is there a reason we don't just use [`authenticate_user!` from devise](https://github.com/heartcombo/devise/blob/8593801130f2df94a50863b5db535c272b00efe1/lib/devise/controllers/helpers.rb#L17) here?

Reply:
At present our behaviour is "if you're not logged in (via any controller `include`ing the `Secured` module) we redirect you to `/` (effectively an interstitial with a "sign in" button) as opposed to `sessions#new` and squirrel away the place you were going to in `session[:redirect_path]` before redirecting you to that place via [`Auth#after_sign_in_path_for(_user)`](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/dde6e35cd4abdc365121120488faf7b094819b54/app/controllers/concerns/auth.rb#L29-L31)

By contrast `authenticate_user!` raises an uncaught Warden error and doesn't appear to have any redirect semantics. Is there a way of being more "idiomatic Devise" than the above while preserving the redirect behaviour?

Reply:
(I suspect understanding what the idiomatic version looks like is key to understanding how we make our locked users work as `active_for_authentication?` is used by the [`activatable` hook](https://github.com/heartcombo/devise/blob/8593801130f2df94a50863b5db535c272b00efe1/lib/devise/hooks/activatable.rb#L7-L11)